.github/workflows/test-and-coverage.yml

```diff
@@ -0,0 +1,111 @@
+name: Build and Test
+
+on: push
+
+jobs:
+  build_and_test:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        arch: [x86_64, arm64]
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: latest-stable
+    - name: Install Dependencies
+      run: |
+        brew install --formula mpv
+        which mpv
+        brew info mpv --formula
+        brew install --formula xcresultparser
+    - name: Setup
+      run: |
+        ./setup.sh
+    - name: Run Tests
+      run: |
+        xcodebuild clean test \
+          -scheme mpvx \
+          -configuration Debug \
+          -derivedDataPath build \
+          -destination 'platform=macOS,arch=${{ matrix.arch }}' \
+          -resultBundlePath artifacts/ResultBundle-${{ matrix.arch }}.xcresult || true
```

Comment:
![reveal-meme](https://github.com/user-attachments/assets/1557448a-ffc0-4913-96b7-c2009eb2bbae)

Reply:
Hmm, seems no way to specify a certain check.
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/monitoring-workflows/adding-a-workflow-status-badge
I will leave it be for now.